spec/std/io/buffered_spec.cr

```diff
@@ -94,6 +94,13 @@ describe "IO::Buffered" do
     io.gets('a', 3).should be_nil
   end
 
+  it "does gets with char and limit without off-by-one" do
+    io = BufferedWrapper.new(MemoryIO.new("test\nabc"))
+    io.gets('a', 5).to_s.size.should eq 5
```

Comment:
I would test what the result string is, for example:

```cr
io.gets('a', 5).should eq("test\n")
```

Same in the specs below